docs/graphs/graph-concepts/graphs-and-indexes.md

```diff
@@ -0,0 +1,105 @@
+---
+title: Optimizing Graphs with Indexing
+sidebar_position: 100
+---
+
+Indexing is a technique that enhances query performance by providing an efficient way to access data in the database. In the context of graph queries, indexing can considerably improve traversals, searches, and other operations involving vertices and edges. Indexes allow the database to quickly locate relevant data without scanning the entire collection.
+
+For more information about indexes and indexing, refer to [Indexing](../../collections/indexing/).
+
+## Key Indexing Aspects for Graph Queries
+
+Here are some important aspects to consider when indexing for graph queries.
+
+### Vertex-Centric Indexes
+
+Vertex-centric indexes optimize queries involving filtering or sorting based on vertex properties. In Macrometa, create a "hash" or "skiplist" index on a vertex collection to expedite queries filtering vertices by specific attributes. For example, in a social network graph, if querying users by their email address is frequent, a "hash" index on the email attribute can enhance query performance.
+
+### Edge-Centric Indexes
+
+Edge-centric indexes optimize queries involving filtering or sorting based on edge properties. In Macrometa, create a "hash", "skiplist", or "persistent" index on an edge collection to expedite queries filtering edges by specific attributes. For example, if querying relationships in a social network graph by a "relationship type" attribute is common, a "hash" index on the relationship type can enhance query performance.
+
+### Improving Traversal Performance
+
+Traversal performance can be impacted by the choice of indexes on vertex and edge collections. Proper indexing can quickly locate traversal starting points and navigate the graph efficiently. Analyze your application's specific traversal patterns to determine the most beneficial indexes.
+
+### Balancing Indexing
+
+When creating indexes, balance the benefits of indexing with the overhead of maintaining them. Indexes improve query performance but consume storage space and require extra maintenance during data modifications. Create indexes selectively, concentrating on the properties and relationships frequently used in your application's queries.
+
+## Choosing the Right Index Type for Graph Performance
+
+Selecting the appropriate index type for your specific use case and query patterns is crucial for enhancing graph performance. Consider the following index types for graph data in Macrometa.
+
+### Hash Index
+
+A hash index is ideal for equality lookups on single attributes. It offers fast document access based on the indexed attribute but does not support range queries or sorting. If your graph queries involve filtering vertices or edges by a specific attribute's exact value, then a hash index can be useful.
```

Comment:
We should add a note that Hash indexes are deprecated and replaced by persistent indexes. I think that we can create them via API call but not via GUI.